components/views/login/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { useState, FormEvent } from 'react'
+import styles from './Login.module.scss'
+import useForm from '../../../utilities/hooks/useForm'
+import Link from 'next/link'
+
+interface FormProps {
+    email: string
+    password: string
+}
+
+const INITIAL_STATE = {
+    email: '',
+    password: ''
+}
+
+const Login = () => {
+    const { input, handleChange, handleReset, clearForm } =
+        useForm(INITIAL_STATE)
+    const [error, setError] = useState({
```

Comment:
Same concept as the initial state. Any time you have some sort of initial state like this, you want to make it a variable, with types. Create the `INITIAL_ERROR_STATE` variable, assign it types, then pass it in instead of the hard typed object here.